pkg/controller/registry/registry_controller.go

```diff
@@ -98,36 +96,20 @@ func (r *ReconcileRegistry) Reconcile(request reconcile.Request) (reconcile.Resu
 	originalRegistry := registry.DeepCopy()
 
 	// Fetch the Instance
-	harbor := &registriesv1alpha1.Instance{}
-	ns := types.NamespacedName{
-		Namespace: registry.Namespace,
-		Name:      registry.Spec.ParentInstance.Name,
-	}
-	err = r.client.Get(ctx, ns, harbor)
-	if errors.IsNotFound(err) {
-		registry.Status = registriesv1alpha1.RegistryStatus{
-			Name:           string(registriesv1alpha1.RegistryStatusPhaseCreating),
-			Message:        "corresponding harbor instance does not exist",
-			LastTransition: &now,
+	harbor, err := internal.FetchReadyHarborInstance(ctx, registry.Namespace, registry.Spec.ParentInstance.Name, r.client)
+	if err != nil {
+		if err == internal.ErrInstanceNotFound(registry.Spec.ParentInstance.Name) {
```

Comment:
Maybe use a type guard, instead? That way you wouldn't have to re-construct the expected error using the parent instance name.

```suggestion
		if _, ok := err.(internal.ErrInstanceNotFound); ok {
```